include/verilated_random.cpp

```diff
@@ -0,0 +1,212 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//*************************************************************************
+//
+// Code available from: https://verilator.org
+//
+// Copyright 2024 by Antmicro Ltd.  This program is free software; you can
+// redistribute it and/or modify it under the terms of either the GNU Lesser
+// General Public License Version 3 or the Perl Artistic License Version 2.0.
+// SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+//
+//=========================================================================
+///
+/// \file
+/// \brief Verilated randomization implementation code
+///
+/// This file must be compiled and linked against all Verilated objects
+/// that use randomization features.
+///
+/// See the internals documentation docs/internals.rst for details.
+///
+//=========================================================================
+
+#include "verilated_random.h"
+
+#include <fstream>
+#include <sstream>
+
+#include <sys/wait.h>
+
+#define HASH_LEN 1
+
+static const char* solvers[] = {
+    "z3 ",
+    "cvc5 --lang=smt2 --incremental ",
+    "cvc4 --lang=smt2 --incremental ",
```

Comment:
I agree about the tests, however for that we will need to introduce a way to choose a solver and not just blindly use the first one from the list.  I thought that we can just try to find a solver best-effort; they are all supposed to be standards-compliant, so I think including the most popular ones is beneficial to the users with one of them already installed in their system.  I also changed the selection process a bit so that less resources are used for quicker fallback.

Reply:
I suspect we'll quickly find there are significant differences that affect us.  I'd suggest we focus on one for now, in order to reduce your work and our support.  Unless you know the reasons (and document) for people to choose one over the other.  What's installed already I don't see as that important, as long as there's a package available for common distros of our primary choice.